nix/docker/k8s/cardano-nodes.yaml

```diff
@@ -0,0 +1,190 @@
+#
+# NAMESPACE=cardano
+# kubectl create namespace $NAMESPACE
+#
+
+# GCE Storage Class
+---
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: cardano-standard-rwo
+provisioner: kubernetes.io/gce-pd
+parameters:
+  type: pd-standard
+  fstype: ext4
+  replication-type: none
+
+# Relay ###############################################################################################################
+
+# Relay Volume Claim
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: relay-data
+  labels:
+    app: cardano
+    pvc: relay-data
+spec:
+  storageClassName: cardano-standard-rwo
+  accessModes: [ ReadWriteOnce ]
+  resources:
+    requests:
+      storage: 10G
+
+# Relay Pod
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: relay
+  labels:
+    app: cardano
+    pod: relay
+spec:
+  containers:
+  - name: relay
+    image: inputoutput/cardano-node:dev
+    imagePullPolicy: Always
+    env:
+    - name: CARDANO_PUBLIC_IP
+      valueFrom:
+        configMapKeyRef:
+          name: relaycfg
+          key:  publicIP
+    - name: CARDANO_TOPOLOGY
+      valueFrom:
+        configMapKeyRef:
+          name: relaycfg
+          key:  topology
+    - name: CARDANO_CUSTOM_PEERS
+      valueFrom:
+        configMapKeyRef:
+          name: relaycfg
+          key:  customPeers
+    - name: CARDANO_UPDATE_TOPOLOGY
+      value: "true"
+    args: [ "run" ]
+    ports:
+      - containerPort: 3001
+    volumeMounts:
+    - name: datavol
+      mountPath: /opt/cardano/data
+  nodeSelector:
+    nodeType: relay
+  volumes:
+  - name: datavol
+    persistentVolumeClaim:
+      claimName: relay-data
+
+# Relay TCP Service
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: relay-np
+  labels:
+    app: cardano
+spec:
+  type: NodePort
+  selector:
+    app: cardano
+    pod: relay
+  ports:
+  - name: tcp
+    port: 3001
+    nodePort: 30010
+
+# Relay TCP Service
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: relay-clip
+  labels:
+    app: cardano
+spec:
+  clusterIP: 10.3.240.100
```

Comment:
Again, portability to other clusters is considered out of scope in the context of this initial PR. 

For the networking perspective, we want the Relay to be reachable for external ingress (i.e. via NodePort 30010). In the topology for the relay, we point to the internal network IP of the block producer. In the topology for the block producer, we point to the internal network IP of the relay (i.e. what you see here)

Reply:
Hey, I'm new to this but wanted to help. Sorry if I'm not understanding some broader scope, but, if it is all internal communication, why using an IP instead of a service name? This is also the case for the `docker-compose` file. Docker / K8S can translate the service name to the matching IP address, without having to manage it explicitly.

Reply:
It's communication across nodes on a private network. With StatefulSets we have the hostname. Indeed, the clusterIP thing shown here is out of date.

Reply:
No problem, just wanted to confirm that it should also be fixed in the docker-compose file. GJ :)